src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -537,32 +537,29 @@ private class WaitDialogProgress : RestoreOperationProgressUI
                 JoinableTaskFactory jtf,
                 CancellationToken token)
             {
-                return await jtf.RunAsync(async () =>
-                {
-                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                    var waitDialogFactory = serviceProvider.GetService<
-                        SVsThreadedWaitDialogFactory, IVsThreadedWaitDialogFactory>();
-
-                    var session = waitDialogFactory.StartWaitDialog(
-                        waitCaption: Resources.DialogTitle,
-                        initialProgress: new ThreadedWaitDialogProgressData(
-                            Resources.RestoringPackages,
-                            progressText: string.Empty,
-                            statusBarText: string.Empty,
-                            isCancelable: true,
-                            currentStep: 0,
-                            totalSteps: 0));
-
-                    return new WaitDialogProgress(session, jtf);
-                });
+                await jtf.SwitchToMainThreadAsync();
+
+                var waitDialogFactory = serviceProvider.GetService<
+                    SVsThreadedWaitDialogFactory, IVsThreadedWaitDialogFactory>();
+
+                var session = waitDialogFactory.StartWaitDialog(
```

Comment:
The logger class is cross-compiled for VS14/VS15 support. Legacy UI used to pop up TWD hence the confusion. This code will be removed later.